Fixed a little warning, because using recent ffmpeg

ebachard opened the pull request:
Is ffmpeg stable ?  :-)

av_register_all() is now deprecated, and I have added a little change to avoid a warning at buildtime.

Use whatever license you want for the change, if you accept the PR.

Thanks !

ebachard committed:
Fixed a little warning, because using recent ffmpeg

hansonLGE commented:
ebachard, Thank you for your suggestion. However, for the latest version(ffmpeg-3.4.2.tar.bz2),  I reviewed av_register_all() defined in avformat.h.  It is not deprecated. Even if it is deprecated, a new register interface should be provided. but  i did not find it.  Can you tell me the version you use?

ebachard commented:
Sorry, I missed to explain : I'm using the git one, after a git pull made 2 days ago only. It means it's probably an extremely recent change, indeed.

FYI, here is what I got in avformat.h:

```
#if FF_API_NEXT
/**
 * Initialize libavformat and register all the muxers, demuxers and
 * protocols. If you do not call this function, then you can select
 * exactly which formats you want to support.
 *
 * @see av_register_input_format()
 * @see av_register_output_format()
 */
attribute_deprecated
void av_register_all(void);

attribute_deprecated
void av_register_input_format(AVInputFormat *format);
attribute_deprecated
void av_register_output_format(AVOutputFormat *format);
#endif

```

ebachard committed:
Problem with avformat_open_input on Windows (as usual), caused by std::wstring instead of std::string or something close

ebachard committed:
Update FFmpegPlayer.cpp

The simple fix to open a file on Windows. The explanation stands in libavutil/mem.c

ebachard committed:
Update FFmpegPlayer.cpp

Reversing the wrong change. But using av_strdup() is really the solution.
